### PR TITLE
Allow the hadoop version and zookeeper version to be overriden

### PR DESCRIPTION
**Pull request by z-york**
This also upgrades the presto-accumulo's hadoop dependency to match the hadoop-apache2 project.

### CODE REVIEW

**Comment by z-york** [comment]
@martint @cberner could you take a look please?

**Comment by electrum** [comment]
Can you explain the motivation for this change?

File: pom.xml [hunk 1]
```diff
@@ -51,6 +51,7 @@
         <dep.slice.version>0.27</dep.slice.version>
         <dep.aws-sdk.version>1.11.30</dep.aws-sdk.version>
         <dep.tempto.version>1.18</dep.tempto.version>
+        <dep.zookeeper.version>3.4.6</dep.zookeeper.version>
```
Comment (electrum) [inline pom.xml]: This isn't needed since it's only used in one place. We only add properties when multiple dependencies need to share the same version.

Comment (z-york) [inline pom.xml]: We are building with our own version of hadoop (through presto-hadoop-apache2) which is built with a zookeeper 3.4.8. In order to pass the dependency check for presto, we need to also override this property when we build. As far as I know, there is no way to override a version at build time unless it is extracted to a property.

**Comment by z-york** [comment]
The reason for this change is because we are building with a custom version of hadoop (through presto-hadoop-apache2) and since we are building with a much later version of hadoop than is bundled with the presto-accumulo (2.7.3 vs 2.2.0) module, we are running into dependency checker errors which are failing our builds. These only started happening since 0.153 when presto-accumulo came in as a module and has prevented our upgrading to a newer version (without this patch). 

Another way this could be accomplished would be having presto-accumulo be based off of presto-hadoop-apache2 instead of just bringing in its own dependencies.

**Comment by electrum** [comment]
Thanks for the explanation. If we can update Accumulo to the `hadoop-apache2` artifact, that will make this easier. Regarding ZooKeeper, it looks like 3.4.9 is the latest version. Would there be any issues with updating everything to use that?

@adamjshook thoughts here?

**Comment by adamjshook** [comment]
@electrum I don't see any issues updating the Accumulo connector to use `hadoop-apache2`.  We're using Hadoop 2.5 and ZooKeeper 3.4.6 and I don't recall any breaking changes in the newest versions.

**Comment by z-york** [comment]
@electrum I will work on getting accumulo to use the hadoop-apache2 artifact. 

For the Zookeeper change, why don't you want to use properties? I've always thought that it was good practice to use properties for dependency versions. It allows people to override with the specific versions that they are using (if they are compatible) and it also provides one place to look for the dependency versions at the top of the file instead of having to look through all the dependencies. It might take a bit for us to upgrade to ZK 3.4.9 (I haven't tried yet though).

**Comment by z-york** [comment]
@electrum https://github.com/prestodb/presto-hadoop-apache2/pull/19 opened to address moving presto-accumulo hadoop dependencies to presto-hadoop-apache2. I will update this PR after that gets in.

**Comment by z-york** [comment]
We ended up upgrading to ZooKeeper 3.4.9, but I still think it is better to abstract this out to a property to avoid these conflicts in the future.

**Comment by electrum** [comment]
I'm closing this now that #7324 has been merged and ZooKeeper is on 3.4.9. If you see any issues or need anything else, please let us know.

**Pull request closed without merging**